sklearn/decomposition/pca.py

```diff
@@ -362,7 +362,7 @@ def _fit(self, X):
             raise TypeError('PCA does not support sparse input. See '
                             'TruncatedSVD for a possible alternative.')
 
-        X = check_array(X, dtype=[np.float64], ensure_2d=True,
+        X = check_array(X, dtype=[np.float32, np.float64], ensure_2d=True,
```

Comment:
Ok so int gets cast to f64 right? I think that's what @GaelVaroquaux wanted too... I'll push a commit. Thx